File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Concepts/VisualEffects/BorderImage/BorderImagePage.cs
using System;
using System.Text;
using System.Data;
using System.Drawing;
using System.ComponentModel;
using System.Collections.Generic;

using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Common.Resources;

namespace CompanionKit.Concepts.VisualEffects.BorderImage
{
    public partial class BorderImagePage : UserControl
    {
        //Constants of BorderImage's parameters
        const int mintBorderOffset = 5;
        const int mintBorderSize = 15;

        public BorderImagePage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles the SelectedIndexChanged event of the objBorderRepeatComboBox control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjBorderRepeatComboBox_SelectedIndexChanged(object sender, EventArgs e)
        {
            if (mobjBorderRepeatComboBox.SelectedIndex > 0)
            {
                //Applies parameter of BorderRepeat to BorderImage visual effect, if such effect already exists
                mobjVisualButton.VisualEffect = new Gizmox.WebGUI.Common.Forms.VisualEffects.BorderImageVisualEffect(new ImageResourceHandle("border.png"), mintBorderOffset, mintBorderOffset, mintBorderOffset, mintBorderOffset, (mobjBorderRepeatComboBox.SelectedIndex-1), (mobjBorderRepeatComboBox.SelectedIndex-1), mintBorderSize, mintBorderSize, mintBorderSize, mintBorderSize);
                mobjVisualButton.Update();
            }
            else { mobjVisualButton.VisualEffect = null; mobjVisualButton.Update(); }
        }
    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Concepts/VisualEffects/BorderImage/BorderImagePage.vb
Namespace CompanionKit.Concepts.VisualEffects.BorderImage

	Public Class BorderImagePage

        'Constants of BorderImage's parameters
        Const mintBorderOffset As Integer = 5
        Const mintBorderSize As Integer = 15

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()

        End Sub

      ''' <summary>
        ''' Handles the SelectedIndexChanged event of the objBorderRepeatComboBox control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub mobjBorderRepeatComboBox_SelectedIndexChanged(sender As Object, e As EventArgs) Handles mobjBorderRepeatComboBox.SelectedIndexChanged
            If mobjBorderRepeatComboBox.SelectedIndex > 0 Then
                'Applies parameter of BorderRepeat to BorderImage visual effect, if such effect already exists
                mobjVisualButton.VisualEffect = New Gizmox.WebGUI.Common.Forms.VisualEffects.BorderImageVisualEffect(New ImageResourceHandle("border.png"), mintBorderOffset, mintBorderOffset, mintBorderOffset, mintBorderOffset, (mobjBorderRepeatComboBox.SelectedIndex - 1), _
                 (mobjBorderRepeatComboBox.SelectedIndex - 1), mintBorderSize, mintBorderSize, mintBorderSize, mintBorderSize)
                mobjVisualButton.Update()
            Else
                mobjVisualButton.VisualEffect = Nothing
                mobjVisualButton.Update()
            End If
        End Sub

	End Class

End Namespace